File: VB/PersonalOrganizer/Common/ViewModel/ISingleObjectViewModel.vb
Imports System
Imports System.ComponentModel.DataAnnotations
Imports System.Linq
Imports System.Linq.Expressions
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.POCO
Imports DevExpress.Mvvm.DataAnnotations
Imports Common.Utils
Imports Common.DataModel
Imports MessageBoxButton = System.Windows.MessageBoxButton
Imports MessageBoxImage = System.Windows.MessageBoxImage
Imports MessageBoxResult = System.Windows.MessageBoxResult
Namespace Common.ViewModel
    Public Interface ISingleObjectViewModel(Of TEntity, TPrimaryKey)
        ReadOnly Property Entity As TEntity
        ReadOnly Property PrimaryKey As TPrimaryKey
    End Interface
    Public Interface IDetailEntityInfo
    End Interface
    Public Class DetailEntityInfo(Of TDetailEntity As Class)
        Implements IDetailEntityInfo
        Private _DetailEntityKey As Object
        Public ReadOnly Property DetailEntityKey As Object
            Get
                Return _DetailEntityKey
            End Get
        End Property
        Public Sub New(ByVal detailEntityKey As Object)
            Me._DetailEntityKey = detailEntityKey
        End Sub
    End Class
End Namespace

File: CS/PersonalOrganizer/Common/ViewModel/ISingleObjectViewModel.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Linq.Expressions;
using DevExpress.Mvvm;
using DevExpress.Mvvm.POCO;
using DevExpress.Mvvm.DataAnnotations;
using PersonalOrganizer.Common.Utils;
using PersonalOrganizer.Common.DataModel;
using MessageBoxButton = System.Windows.MessageBoxButton;
using MessageBoxImage = System.Windows.MessageBoxImage;
using MessageBoxResult = System.Windows.MessageBoxResult;

namespace PersonalOrganizer.Common.ViewModel {
    public interface ISingleObjectViewModel<TEntity, TPrimaryKey> {
        TEntity Entity { get; }
        TPrimaryKey PrimaryKey { get; }
    }
    public interface IDetailEntityInfo { }
    public class DetailEntityInfo<TDetailEntity> : IDetailEntityInfo
        where TDetailEntity : class {
        public object DetailEntityKey { get; private set; }
        public DetailEntityInfo(object detailEntityKey) {
            this.DetailEntityKey = detailEntityKey;
        }
    }
}